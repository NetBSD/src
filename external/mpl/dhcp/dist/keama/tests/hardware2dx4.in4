# simplified hardware data expression

# hardware type class
class "ethernet" {
    match if substring(hardware, 0, 1) = encode-int(1, 8);
}

# ethernet address superclass
class "ethernet-address" {
    match substring(hardware, 1, 6);
}

subclass "ethernet-address" 00:0B:FD:32:E6:FA { }
