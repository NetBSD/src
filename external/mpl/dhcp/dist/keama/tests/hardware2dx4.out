{
  # simplified hardware data expression
  # hardware type class
  "Dhcp4": {
    "client-classes": [
      {
        "name": "ethernet",
        /// from: match if (substring(hardware, 0, 1)) = (encode-int(1, 8))
        "test": "substring(pkt4.htype,-1,all) == '\u0001'"
      },
      # ethernet address superclass
      /// match: substring(hardware, 1, 6)
      {
        "name": "ethernet-address"
      },
      /// subclass selector 0x0x000bfd32e6fa
      {
        "name": "sub#ethernet-address#0",
        /// from: match substring(hardware, 1, 6)
        /// data: 0x000bfd32e6fa
        "test": "substring(pkt4.mac,0,6) == 0x000bfd32e6fa"
      }
    ]
  }
}
